pkg/asset/agent/manifests/clusterimageset.go

```diff
@@ -67,5 +103,47 @@ func (a *ClusterImageSet) Load(f asset.FileFetcher) (bool, error) {
 	}
 	a.Config = clusterImageSet
 
+	if err = a.finish(); err != nil {
+		return false, err
+	}
 	return true, nil
 }
+
+func (a *ClusterImageSet) finish() error {
+
+	if a.Config == nil {
+		return errors.New("missing configuration or manifest file")
+	}
+
+	if err := a.validateClusterImageSet().ToAggregate(); err != nil {
+		return errors.Wrapf(err, "invalid ClusterImageSet configuration")
+	}
+
+	return nil
+}
+
+func (a *ClusterImageSet) validateClusterImageSet() field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if err := a.validateReleaseVersion(); err != nil {
+		allErrs = append(allErrs, err...)
+	}
+
+	return allErrs
+}
+
+func (a *ClusterImageSet) validateReleaseVersion() field.ErrorList {
+
+	var allErrs field.ErrorList
+
+	fieldPath := field.NewPath("Spec", "ReleaseImage")
+
+	releaseImage := &releaseimage.Image{}
+	releaseImage.Generate(asset.Parents{})
+
+	if a.Config.Spec.ReleaseImage != releaseImage.PullSpec {
```

Comment:
I don't know that we need to be _this_ strict. I am happy if the minor version (4.y) matches. It may be that we don't have enough information to verify in that case?

My guess is that this will be fine if you either download the installer from cloud.redhat.com and use the hosted payload, or if you extract the installer from a mirror. But it may prevent you from using the downloaded installer with a mirrored payload. I think that's acceptable iff this is already the case for IPI.

Reply:
This approach AFAIK continues to honor the already existing one in the standard installer, where it's not possible to install a release version different from the one hard-codeed in the binary (unless overridden through the env var). For the mirrored payload I guess that this approach should work with what @bfournie is implementing in https://github.com/openshift/installer/pull/5992.
(and yes, unfortunately we don't have enough information to verify if it's a minor version or not)

Reply:
OK, I think this is fine. The built-in pullspec appears to always be a hash anyway. Let's go ahead.